### PR TITLE
Revert "auto deactivate extension for webui >= 1.5.0"

### PR DESCRIPTION
**Pull request by SELEKCJONER**
Reverts KohakuBlueleaf/a1111-sd-webui-lycoris#61

### CODE REVIEW

**Comment by SELEKCJONER** [comment]
why would forcing me or anyone else to use builtin extension be a good thing obsolecence notice would be better but not forcefull deactivation

**Comment by KohakuBlueleaf** [comment]
> why would forcing me or anyone else to use builtin extension be a good thing obsolecence notice would be better but not forcefull deactivation

Bcuz lycoris ext didn't supprt sd-webui>=1.5.0 and may caused bugs or unexpected behaviour

It is not supposed to be used in sd-webui>=1.5.0
And all its features and functionality are now integrated into sd-webui bultin lora ext

And new things like norm module or full module with bias are directly implemented into built-in lora

**Comment by KohakuBlueleaf** [comment]
TL; DR;

Just take built-in lora in sd-webui>=1.5.0 as new lycoris ext

**Comment by SELEKCJONER** [comment]
I understand that but still i think that there should be at least some option to disable automatic deactivation with incompatibility notice instead of forcefull deactivation

**Comment by KohakuBlueleaf** [comment]
> I understand that but still i think that there should be at least some option to disable automatic deactivation with incompatibility notice instead of forcefull deactivation

Why?
Bcuz <lyco>? You can use <lyco> in built-in lora after 1.5.0

**Comment by SELEKCJONER** [comment]
Because new sd webui lora extension doesnt segregate Lycoris and lora which i can do by using this extension. Builtin extension shows LyCORIS folder contents in Lora tab instead of its own tab and i havent tested this but i am preety sure it forces me to use <lora:filename:1> instead of <lyco:filename:1> but i did not even try to test it and immieadeately disabled LyCORIS folder content merging with lora tab in a fork.

**Comment by w-e-w** [comment]
why do you want to use `lyco:filename:1` `lyco` is an alias of `lora`
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/12668#issuecomment-1684941138

**Comment by SELEKCJONER** [comment]
Because that is what i have in many of my saved prompts where Lycoris models are used and i do not want to change it and i segregate LyCORIS models and Lora models in their respective folders with expectation for them to be put in separate tabs which webui is not doing and i do not have sufficient programming knowledge to implement such segregation in my fork.

**Comment by w-e-w** [comment]
`lyco:filename:1` still works, it's back with compatible
after 1.5.0 the end user don't need to care if you're using a normal lora or a lyco model
if you found out you owed prompt does not produce the same result then that would suggest that there is a bug

**Comment by SELEKCJONER** [comment]
Last time i did try to use new lora extension with lyco model i was in fact getting different results where <lyco:filename:1> was giving me the better result it didnt segregate them in separate tabs either and showed LyCORIS folder content in Lora tab instead also thank you for pointing out that webui will threat <lyco:filename:1> as <lora:filename:1> i will delete alias in my fork to make sure that i am using this extension with auto deactivate reverted instead

**Comment by CasanovaSan** [comment]
Bring back the extension, i want my shit in ordered, i want to know that my lycoris are in the lycoris folder and lora in the lora folder

**Comment by VL4DST3R** [comment]
If you wanted to let the users know about the extension now being obsolete how about you properly notify them inside the LyCO tab with a popup or at the very least here on git instead of pulling crap like this?

**Comment by KohakuBlueleaf** [comment]
> If you wanted to let the users know about the extension now being obsolete how about you properly notify them inside the LyCO tab with a popup or at the very least here on git instead of pulling crap like this?

It is hard to say this is "obsolete" since lot of user are still using old version sd-webui.
This extension now is only for old version sd-webui (kind like the "future" pack in python)
The thing is, lycoris ext are just uncompatible with new sd-webui and will cause lot of bugs (especially on 1.6.0, I don't think it can even work properly)

After 1.5.0 lycoris ext just should not appeared.

If you want any "functionality" or "feature" that related to how you use lycoris ext before these changes, it is a feature request on sd-webui side not lycoris ext side.

**Comment by KohakuBlueleaf** [comment]
> Bring back the extension, i want my shit ordered, i want to know that my lycoris are in the lycoris folder and lora in the lora folder

1, you can still put lora in lora folder, lycoris in lycoris folder

2, lora is subset of lycoris, basically you can't distinguish "which is lora which is lycoris", If you want to say "lycoris cannot be supported by old sd webui" how old? Sd webui have added different compatibility before merge lycoris.

3, if you just want separated tab for separated folder, I can make a option in sd-webui to let you enable LyCORIS tab.

4, If you just want totally separated system to handle very complex weight backup mechanism with lot of edge cases and issues. Fork it and maintain it, this is open sourced, no one can stop you.

**Comment by KohakuBlueleaf** [comment]
> Last time i did try to use new lora extension with lyco model i was in fact getting different results where lyco:filename:1 was giving me the better result it didnt segregate them in separate tabs either and showed LyCORIS folder content in Lora tab instead also thank you for pointing out that webui will threat lyco:filename:1 as lora:filename:1 i will delete alias in my fork to make sure that i am using this extension with auto deactivate reverted instead

We have tested it that new lora ext(on new sd webui) and old lyco ext(on old sd webui) will generate identical result
So the thing you met is a "expected bug". But the bug just give you better result

Better result is not equal to correct result.

**Comment by SELEKCJONER** [comment]
Okay if Lycoris extension cannot work but you are able to make separete tab for lycoris models for sorting purposes then how will you stop webui from showing LyCORIS folder content in Lora tab. I have identified which line causes it and its line number 402 in file stable-diffusion-webui/extensions-builtin/Lora/networks.py 

candidates += list(shared.walk_files(shared.cmd_opts.lyco_dir_backcompat, allowed_extensions=[".pt", ".ckpt", ".safetensors"]))

**Comment by KohakuBlueleaf** [comment]
> > 3, if you just want separated tab for separated folder, I can make a option in sd-webui to let you enable LyCORIS tab.
> 
> yes, that's what i personally want, i just want to have more order

I will add this into sd-webui built-in lora directly
with a option

**Comment by KohakuBlueleaf** [comment]
> Okay if Lycoris extension cannot work but you are able to make separete tab for lycoris models for sorting purposes then how will you stop webui from showing LyCORIS folder content in Lora tab. I have identified which line causes it and its line number 402 in file stable-diffusion-webui/extensions-builtin/Lora/networks.py
> 
> candidates += list(shared.walk_files(shared.cmd_opts.lyco_dir_backcompat, allowed_extensions=[".pt", ".ckpt", ".safetensors"]))

Just FYI, sd-webui is a open sourced project which accept PRs

**Comment by VL4DST3R** [comment]
@KohakuBlueleaf 
Thank you for the clarification. Still, a warning would have been nice instead of worrying my lyco models just vanished. For a bit of context, I've noticed a certain amount of disdain from some a1111 extension developers when it came to lyco, so my first reaction was to freak out when i saw them gone entirely.

> The thing is, lycoris ext are just uncompatible with new sd-webui and will cause lot of bugs (especially on 1.6.0, I don't think it can even work properly)

What do you mean exactly? I couldn't find any mention of a 1.6 webui version for a1111, unless I'm missing something?

**Comment by w-e-w** [comment]
he mentally upcoming version of webui 1.6.0

**Comment by VL4DST3R** [comment]
> he meant the upcoming version of webui 1.6.0

Yeah i figured that much but i couldn't find any mention of 1.6 on a1111 git or even broadly on the web, i was curious to see what changes it brings.

**Comment by KohakuBlueleaf** [comment]
> > he meant the upcoming version of webui 1.6.0
> 
> Yeah i figured that much but i couldn't find any mention of 1.6 on a1111 git or even broadly on the web, i was curious to see what changes it brings.

You can check the dev branch
Should have a rough changelog now
Or directly clone it (or git switch/checkout dev)

**Comment by dhwz** [comment]
> Bring back the extension, i want my shit ordered, i want to know that my lycoris are in the lycoris folder and lora in the lora folder

you know there are subfolders supported for every extra network tab?

**Comment by dhwz** [comment]
> Okay if Lycoris extension cannot work but you are able to make separete tab for lycoris models for sorting purposes then how will you stop webui from showing LyCORIS folder content in Lora tab. I have identified which line causes it and its line number 402 in file stable-diffusion-webui/extensions-builtin/Lora/networks.py
> 
> candidates += list(shared.walk_files(shared.cmd_opts.lyco_dir_backcompat, allowed_extensions=[".pt", ".ckpt", ".safetensors"]))

same for you why don't use folders? there is no raeson for a 2nd tab on current WebUI, it supports multiple subfolders for every extra network tab, including filtering.

**Comment by SELEKCJONER** [comment]
> > Okay if Lycoris extension cannot work but you are able to make separete tab for lycoris models for sorting purposes then how will you stop webui from showing LyCORIS folder content in Lora tab. I have identified which line causes it and its line number 402 in file stable-diffusion-webui/extensions-builtin/Lora/networks.py
> > candidates += list(shared.walk_files(shared.cmd_opts.lyco_dir_backcompat, allowed_extensions=[".pt", ".ckpt", ".safetensors"]))
> 
> same for you why don't use folders? there is no raeson for a 2nd tab on current WebUI, it supports multiple subfolders for every extra network tab, including filtering.

I assume that you mean the extra buttons that appear when inside of Lora or Lycoris folder is another folder. That approach only gives an option to show content of a specific subfolder located in lora or lycoris folder and does not make a separate tab which is what lycoris extension was doing and builtin lora extension does not do.

The reason why i want separate tab is to clearly separate lycoris models from lora models and make webui look better without filter buttons but with separate lycoris tab just like it always used to be done.

**Comment by dhwz** [comment]
But it was only like that because @KohakuBlueleaf hacked it in like that as extension.
There is no difference between a Lora and Lycoris, so why do you want to still separate them (even Civitai could now just list them all under Lora).

**Comment by SELEKCJONER** [comment]
> But it was only like that because @KohakuBlueleaf hacked it in like that as extension. There is no difference between a Lora and Lycoris model, so why do you want to still separate them, even Civitai could (or maybe also will) now just list them all under Lora.
> 
> I've made a proposal to @KohakuBlueleaf that we may mark them with a label/icon like on Civitai

Because regardles of them serving the same purpose they are not the same models and work in a different way and built in lora extension just handles these cases.
